server/src/main/java/org/opensearch/rest/extensions/RestSendToExtensionAction.java

```diff
@@ -78,23 +84,39 @@ public String getName() {
     public RestSendToExtensionAction(
         RegisterRestActionsRequest restActionsRequest,
         DiscoveryExtensionNode discoveryExtensionNode,
-        TransportService transportService
+        TransportService transportService,
+        ActionModule actionModule
     ) {
         this.pathPrefix = "/_extensions/_" + restActionsRequest.getUniqueId();
         RestRequest.Method method;
         String path;
 
         List<Route> restActionsAsRoutes = new ArrayList<>();
         for (String restAction : restActionsRequest.getRestActions()) {
-            int delim = restAction.indexOf(' ');
+            Optional<String> name = Optional.empty();
+            String[] parts = restAction.split(" ");
+            if (parts.length < 2) {
+                throw new IllegalArgumentException("REST action must contain at least a REST method and route");
+            }
             try {
-                method = RestRequest.Method.valueOf(restAction.substring(0, delim));
-                path = pathPrefix + restAction.substring(delim).trim();
+                method = RestRequest.Method.valueOf(parts[0].trim());
+                path = pathPrefix + parts[1].trim();
+                if (parts.length > 2) {
+                    name = Optional.of(parts[2].trim());
+                }
             } catch (IndexOutOfBoundsException | IllegalArgumentException e) {
                 throw new IllegalArgumentException(restAction + " does not begin with a valid REST method");
             }
             logger.info("Registering: " + method + " " + path);
-            restActionsAsRoutes.add(new Route(method, path));
+            if (name.isPresent()) {
+                NamedRoute nr = new NamedRoute(method, path, name.get());
+                restActionsAsRoutes.add(nr);
+                actionModule.getDynamicRouteRegistry().registerDynamicRoute(nr, this);
```

Comment:
So ... you just need `DynamicRouteRegistery` now, right? ;-)

Reply:
I folded this into `DynamicActionRegistry` so there's one dynamic registry which contains native + plugin transport actions, dynamic transport actions and now dynamically registered routes. I also added 2 tests around registering routes with path parameters.

Reply:
Yes, I updated this now so that its only passing around the dynamic registry